list-agreements.lua

```diff
@@ -1,120 +1,222 @@
-    religion_id = agr.details[0].data.Location.deity_data.Religion
-    religion_name = dfhack.TranslateName(df.global.world.entities.all[religion_id].name, true)
-    return religion_name
+function get_agr_party_name(agr)
+    --assume party 0 is guild/order, 1 is local government as siteid = playerfortid
+    local party_id = agr.parties[0].entity_ids[0]
+    local party_name = dfhack.TranslateName(df.global.world.entities.all[party_id].name, true)
+    if not party_name then
+        party_name = 'An Unknown Entity or Group'
+    end
+    return party_name
 end
 
-function is_satisfied(agr)
-    satisfied = agr.flags.convicted_accepted
-    return satisfied
+function get_deity_name(agr)
+    local religion_id = agr.details[0].data.Location.deity_data.Religion
+    local deities = df.global.world.entities.all[religion_id].relations.deities
+    if #deities == 0 then return 'An Unknown Deity' end
+    return dfhack.TranslateName(df.global.world.history.figures[deities[0]].name,true)
 end
 
-function is_denied(agr)
-    denied = agr.flags.petition_not_accepted
-    return denied
+--get resolution status, and string
+function is_resolved(agr)
+    local res = false
+    local res_str = 'outstanding'
+
+    if agr.flags.convicted_accepted then
+        res = true
+        res_str = 'satisfied'
+    elseif agr.flags.petition_not_accepted then
+        res = true
+        res_str = 'denied'
+    elseif get_petition_age(agr)[1] ~= 0 then
+        res = true
+        res_str = 'expired'
+    end
+    return res,res_str
 end
 
-function generate_output_guild(agr)
-    if is_satisfied(agr) == true then
-        print("Establish a "..get_location_name(agr).." for the "..get_guildhall_profession(agr).." guild, as agreed on "..get_petition_date(agr).." (satisfied)")
-    elseif is_denied(agr) == true then
-        print("Establish a "..get_location_name(agr).." for the "..get_guildhall_profession(agr).." guild, as agreed on "..get_petition_date(agr).." (denied)")
+--universal handler
+function generate_output(agr,loctype)
+    local loc_name = get_location_name(agr.details[0].data.Location.tier,loctype)
+    local agr_age = get_petition_age(agr)
+    local output_str = 'Establish a '.. loc_name..' for "'..get_agr_party_name(agr)
+    local resolved, resolution_string = is_resolved(agr)
+
+    if loctype == df.abstract_building_type.TEMPLE then
+        output_str = output_str..'" worshiping "'..get_deity_name(agr)..',"'
+    elseif loctype == df.abstract_building_type.GUILDHALL then
+        output_str = output_str..'", a '..get_guildhall_profession(agr)..' guild,'
     else
-        print("Establish a "..get_location_name(agr).." for the "..get_guildhall_profession(agr).." guild, as agreed on "..get_petition_date(agr))
+        print("Agreement with unknown org")
+        return
+    end
+
+    output_str = output_str..'\n\tas agreed on '..get_petition_date(agr)..'. \t'..agr_age[1]..'y, '..agr_age[2]..'m, '..agr_age[3]..'d ago'
```

Comment:
`\t` appears to be inserting ascii-symbol
![image](https://user-images.githubusercontent.com/46378533/226090255-89a1b68f-502f-4609-a35f-f27376ff5557.png)

Reply:
Resolved in #640